src/com/owncloud/android/files/services/FileUploader.java

```diff
@@ -941,6 +949,34 @@ public void uploadFile(String uploadKey) {
 
                 mUploadsStorageManager.updateDatabaseUploadResult(uploadResult, mCurrentUpload);
 
+                if (!uploadResult.isSuccess() && uploadResult.getException() != null) {
+
+                    // Check network availabality
+                    if (!ConnectivityUtils.isNetworkActive(getApplicationContext())) {
+
+                        // Schedule future retry of failed upload due to network error from Android 5
+                        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
+
+                            ComponentName mServiceComponent = new ComponentName(this,
+
+                                    RetryUploadJobService.class);
+
+                            JobInfo.Builder builder;
+
+                            builder = new JobInfo.Builder(jobId, mServiceComponent);
+                            // require unmetered network
+                            builder.setRequiredNetworkType(JobInfo.NETWORK_TYPE_ANY);
+                            PersistableBundle extras = new PersistableBundle();
+                            extras.putString(KEY_FILE, mCurrentUpload.getRemotePath());
```

Comment:
The keys for the extras should be the same for downloads and uploads. They mean the same -  account **name** and file **remote path** .